netbox/core/models/jobs.py

```diff
@@ -197,25 +199,34 @@ def terminate(self, status=JobStatusChoices.STATUS_COMPLETED, error=None):
         job_end.send(self)
 
     @classmethod
-    def enqueue(cls, func, instance, name='', user=None, schedule_at=None, interval=None, **kwargs):
+    def enqueue(cls, func, instance=None, name='', user=None, schedule_at=None, interval=None, immediate=False, **kwargs):
```

Comment:
I think it might be cleaner to break this into a base function with two other functions enqueue and run? to separate out the immediate keyword and schedule_at / interval.   In the immediate case you are not really enqueueing anything. I'm not sure interval is supported when using Immediate?  It is not used by the management command.

Reply:
There was an earlier discussion about naming the parameter and the consequences. You could split this into two methods, both reusing a third internal `_create()` method to make the parameters and behavior a bit clearer to the user. This would also require an additional proxy method in `BackgroundJob`. I'd like to leave the decision to @jeremystretch.